### PR TITLE
Add Networking Guide

### PR DESCRIPTION
**Pull request by geoffcline**
*Description of changes:*
Add an entirely new guide covering EKS Networking.

Thank you to @sheetaljoshi for authoring this content! 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by geoffcline** [comment]
preview will be here, currently updating https://network-bpg.gcline.us

File: content/networking/custom-networking/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+The network architecture used in this blog post implementation follows the recommendations under [Enable communication between overlapping networks](https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-scenarios.html#private-nat-overlapping-networks) in Amazon VPC documentation. As demonstrated in this blog post, you may expand the usage of private NAT Gateway in conjunction with RFC6598 addresses to manage customers' private IP exhaustion issues. The EKS clusters, worker nodes are deployed in the non-routable 100.64.0.0/16 VPC secondary CIDR range, whereas the private NAT gateway, NAT gateway are deployed to the routable RFC1918 CIDR ranges. The blog explains how a transit gateway is used to connect VPCs in order to facilitate communication across VPCs with overlapping non-routable CIDR ranges. For use cases in which EKS resources in a VPC's non-routable address range need to communicate with other VPCs that do not have overlapping address ranges, customers have the option of using VPC Peering to interconnect such VPCs. This method could provide potential cost savings as all data transit within an Availability Zone via a VPC peering connection is now free.
```
Comment (jicowan) [inline content/networking/custom-networking/index.md:68]: I think the link in this paragraph needs to be fixed.

File: content/networking/prefix-mode/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+During worker node initialization, the VPC CNI assigns one or more prefixes to the primary ENI. The CNI pre-allocates a prefix for faster pod startup by maintaining a warm pool. The number of prefixes to be held in warm pool can be controlled by setting environment variables.
+
+* `WARM_PREFIX_TARGET`, the number of prefixes to be allocated in excess of current need.
+* `WARM_IP_TARGET`, the number of IP addresses to be allocated in excess of current need.
+* `MINIMUM_IP_TARGET`, the minimum number of IP addresses to be available at any time.
+* `WARM_IP_TARGET` and `MINIMUM_IP_TARGET` if set will override `WARM_PREFIX_TARGET`.
+
+As more Pods scheduled additional prefixes will be requested for the existing ENI. First, the VPC CNI attempts to allocate a new prefix to an existing ENI. If the ENI is at capacity, the VPC CNI attempts to allocate a new ENI to the node. New ENIs will be attached until the maximum ENI limit (defined by the instance type) is reached. When a new ENI is attached, ipamd will allocate one or more prefixes needed to maintain the WARM_PREFIX_TARGET, WARM_IP_TARGET and MINIMUM_IP_TARGET setting.
```
Comment (jicowan) [inline content/networking/prefix-mode/index.md:20]: Please use code format for the ENVs in this paragraph.

File: content/networking/sgpp/index.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+**Deploying NodeLocal DNSCache**
+
+When using security groups for pods, configure standard mode to support Pods that use [NodeLocal DNSCache](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/). NodeLocal DNSCache improves Cluster DNS performance by running a DNS caching agent on cluster nodes as a DaemonSet. This will help the pods that have the highest DNS QPS requirements to query local kube-dns/CoreDNS having a local cache, which will improve the latency.
+
+NodeLocal DNSCache is not supported in strict mode as all network traffic, even to the node, enters the VPC. 
+
+**Supporting Kubernetes Network Policy**
+
+We recommend using standard enforcing mode when using network policy with Pods that have associated security groups.
+We strongly recommend to utilize security groups for Pods to limit network-level access to AWS services that are not part of a cluster. Consider network policies to restrict network traffic between Pods inside a cluster, often known as East/West traffic.
```
Comment (jicowan) [inline content/networking/sgpp/index.md:75]: Need a CR before this paragraph.

**Review by jicowan — CHANGES_REQUESTED** [review]
Found a few formatting issues. Can you fix those and resubmit. Thanks!

File: content/networking/custom-networking/index.md (hunk 1 above)
Comment (geoffcline) [reply content/networking/custom-networking/index.md:68]: is this the correct post? https://aws.amazon.com/blogs/networking-and-content-delivery/how-to-solve-private-ip-exhaustion-with-private-nat-solution/

Comment (jicowan) [reply content/networking/custom-networking/index.md:68]: I think the link was right. There may have been a space in between the square bracket and the parenthesis.

**Review by jicowan — APPROVED** [review]
/LGTM

**Pull request merged**